src/awkward/_v2/contents/bytemaskedarray.py

```diff
@@ -325,32 +325,40 @@ def _localindex(self, axis, depth):
             )
             return out2._simplify_optiontype()
 
-    def toIndexedOptionArray(self):
-        nplike = self._mask.nplike
-        index = ak._v2.index.Index64.empty(len(self._mask), nplike)
-        self._handle_error(
-            nplike[
-                "awkward_ByteMaskedArray_toIndexedOptionArray",
-                index.dtype.type,
-                self._mask.dtype.type,
-            ](
-                index.to(nplike),
-                self._mask.to(nplike),
-                len(self._mask),
-                self._valid_when,
-            )
-        )
-        return ak._v2.contents.IndexedOptionArray(
-            index,
-            self._content,
-            self._identifier,
-            self._parameters,
```

Comment:
At first, I was worried about this function disappearing, but then I saw that `toIndexedOptionArray`/`toIndexedOptionArray64` is a duplication in the main branch. Thanks for removing it here!